presto-prometheus/src/main/java/io/prestosql/plugin/prometheus/PrometheusSplitManager.java

```diff
@@ -0,0 +1,156 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.prometheus;
+
+import io.prestosql.spi.connector.ConnectorSession;
+import io.prestosql.spi.connector.ConnectorSplit;
+import io.prestosql.spi.connector.ConnectorSplitManager;
+import io.prestosql.spi.connector.ConnectorSplitSource;
+import io.prestosql.spi.connector.ConnectorTableHandle;
+import io.prestosql.spi.connector.ConnectorTransactionHandle;
+import io.prestosql.spi.connector.FixedSplitSource;
+import io.prestosql.spi.connector.TableNotFoundException;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.message.BasicNameValuePair;
+
+import javax.inject.Inject;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static java.util.Objects.requireNonNull;
+
+public class PrometheusSplitManager
+        implements ConnectorSplitManager
+{
+    static final long DAYS_IN_A_YEAR = 365;
+    static final long DAYS_IN_A_WEEK = 7;
+    private final PrometheusClient prometheusClient;
+
+    @Inject
+    public PrometheusSplitManager(PrometheusClient prometheusClient)
+    {
+        this.prometheusClient = requireNonNull(prometheusClient, "client is null");
+        requireNonNull(prometheusClient.config.getQueryChunkSizeDuration(), "query-chunk-size-duration is null");
+        requireNonNull(prometheusClient.config.getMaxQueryRangeDuration(), "max-query-range-duration is null");
+    }
+
+    @Override
+    public ConnectorSplitSource getSplits(
+            ConnectorTransactionHandle transaction,
+            ConnectorSession session,
+            ConnectorTableHandle connectorTableHandle,
+            SplitSchedulingStrategy splitSchedulingStrategy)
+    {
+        PrometheusTableHandle tableHandle = (PrometheusTableHandle) connectorTableHandle;
+        PrometheusTable table = prometheusClient.getTable(tableHandle.getSchemaName(), tableHandle.getTableName());
+
+        // this can happen if table is removed during a query
+        if (table == null) {
+            throw new TableNotFoundException(tableHandle.toSchemaTableName());
+        }
+        List<ConnectorSplit> splits = new ArrayList<>();
+        generateTimesForSplits(Instant.now().toEpochMilli()).forEach(time -> {
+            try {
+                splits.add(new PrometheusSplit(buildQuery(prometheusClient.config.getPrometheusURI(), time, table.getName())));
+            }
+            catch (URISyntaxException e) {
```

Comment:
When does this occur?

Reply:
on `buildQuery` returning a non-conforming URI for whatever reason, but it will not be due to what it has for `config.getPrometheusURI()` since that will already have thrown if there is a URISyntaxException on that.

Reply:
Right, so I'm just curious if you've hit this exception before, or if we've just got the exception handler "just in case". Trying to decide if we should add comments on when this could occur if we knew of any cases already.

Reply:
We just have it because `new URI(String)` usage throws it and we have to catch/throw it.